frontend/heir/backends/openfhe/backend.py

```diff
@@ -0,0 +1,227 @@
+
+  def __init__(self, compilation_result: CompilationResult):
+    self.compilation_result = compilation_result
+
+  def setup(self):
+    # TODO (#1119): Rethink the server/client split
+    # TODO (#1162) : Fix "ImportError: generic_type: type "PublicKey" is already registered!" when doing setup twice. (Required to allow multiple compilations in same python file)
+    self.crypto_context = self.compilation_result.setup_funcs[
+        "generate_crypto_context"
+    ]()
+    self.keypair = self.crypto_context.KeyGen()
+    # Really, "configure" is just setting up non-public key material, and
+    # all public parameter configuration is done on
+    # generate_crypto_context.
+    self.compilation_result.setup_funcs["configure_crypto_context"](
+        self.crypto_context, self.keypair.secretKey
+    )
+
+  def decrypt_result(self, result, *, crypto_context=None, secret_key=None):
+    return self.compilation_result.result_dec_func(
+        crypto_context or self.crypto_context,
+        result,
+        secret_key or self.keypair.secretKey,
+    )
+
+  def __getattr__(self, key):
+    if key == "crypto_context":
+       msg = f"HEIR Error: Please call {self.compilation_result.func_name}.setup() before calling {self.compilation_result.func_name}.encrypt/eval/decrypt";
+       init(autoreset=True)
+       print(Fore.RED + Style.BRIGHT + msg)
+       raise RuntimeError(msg)
+
+    if key.startswith("encrypt_"):
+      # TODO (#1162): Prevent people from doing a = enc_x, b = enc_y for foo(x,y) but then calling foo(b,a)!
+      # TODO (#1119): expose ctxt serialization in python
+      arg_name = key[len("encrypt_") :]
+      fn = self.compilation_result.arg_enc_funcs[arg_name]
+
+      def wrapper(arg, *, crypto_context=None, public_key=None):
+        return fn(
+            crypto_context or self.crypto_context,
+            arg,
+            public_key or self.keypair.publicKey,
+        )
+
+      return wrapper
+
+    if key == self.compilation_result.func_name or key == "eval":
+      # TODO (#1162): Consider hardening the eval function against handcrafted ciphertexts with wrong metadata being passed in?
+      fn = self.compilation_result.main_func
+
+      def wrapper(*args, crypto_context=None):
+        return fn(crypto_context or self.crypto_context, *args)
+
+      return wrapper
+
+    # if key == "func":
+    #   return self.compilation_result.func
+
+    try:
+      getattr(self.compilation_result.module, key)
+    except AttributeError:
+      raise AttributeError(f"Attribute {key} not found")
+
+
+class OpenFHEBackend(BackendInterface):
+    def __init__(self, openfhe_config: OpenFHEConfig):
+        self.openfhe_config = openfhe_config
+        self.heir_opt_options = []
+
+    def run_backend(self, workspace_dir, heir_opt, heir_translate, func_name, arg_names, secret_args, heir_opt_output, debug):
+        # Initialize Colorama for error and debug messages
+        init(autoreset=True)
```

Comment:
nit: The comment would not be needed (more self-documenting) if instead you did this:

```
import colorama
from colorama import Fore, Style

...

    colorama.init(...)
```

Also internal Google style guide does not allow importing barewords, so I will likely have to patch this internally to be

```
import colorama

Fore = colorama.Fore
Style = colorama.Style
```